WhateverGreen/kern_weg.hpp

```diff
@@ -92,11 +92,13 @@ class WEG {
 	 *  APPLBKL_OFF     disables AppleBacklight patches.
 	 *  APPLBKL_ON      enforces AppleBacklight patches.
 	 *  APPLBKL_DETECT  enables AppleBacklight patches for IGPU-only non-Apple setups.
+	 *  APPLBKL_NAVI10  enables AppleBacklight patches for AMD Navi10 PWM backlight control.
 	 */
 	enum BacklightPatchMode {
 		APPLBKL_OFF    = 0,
 		APPLBKL_ON     = 1,
-		APPLBKL_DETECT = 2
+		APPLBKL_DETECT = 2,
+		APPLBKL_NAVI10 = 3
```

Comment:
Are we too lazy to (additionally?) do the autpdetect by device-id? Or is it not enough/unreliable?

Reply:
hi @vit9696, I encountered some difficulties when trying to autodetect this. There are two conditions that need to enable this patch, first is an external AMD Navi10 graphic card, second is the internal display direct-wired to this external Navi10 card. I can auto-detect the first condition. But I'm stuck at auto-detect the second one. My first attempt is to try to read the parent entry of the AppleBacklightDisplay to check if it is an AMD Navi10 framebuffer, but the display is not initialized when executing WEG:init. So I have to check this condition inside or after setDisplay function. And I have to call lilu.onKextLoadForce inside or after setDisplay. And there's also a problem that WEG will call kextBacklight.switchOff() when APPLBKL_DETECT && extNum > 0. And when it's trying to switchOff(), the display is not initialized as well.

Reply:
Right, sounds safer not to autodetect then to be honest.